From 0df5807ead5570ab662db0e9d36e6e019b2cd850 Mon Sep 17 00:00:00 2001
From: Doug Goldstein <cardoe@cardoe.com>
Date: Thu, 6 Nov 2025 16:11:55 -0600
Subject: [PATCH] fix: local_link_connection inspection hook does not fail on
 missing port

The code intended to not fail when a Port object was missing but to
instead skip the handling of that node but instead there is an unhandled
exception because the get_by_address() method throws PortNotFound
instead of returning None when a port couldn't be found.

Change-Id: I04dfa09ada7e6a9d22ba16051cb5737daf3bc668
Signed-off-by: Doug Goldstein <cardoe@cardoe.com>
---
 .../modules/inspector/hooks/local_link_connection.py        | 6 ++++--
 ...t-hook-local-link-connection-crash-394edb1c35354968.yaml | 6 ++++++
 2 files changed, 10 insertions(+), 2 deletions(-)
 create mode 100644 releasenotes/notes/inspect-hook-local-link-connection-crash-394edb1c35354968.yaml

diff --git a/ironic/drivers/modules/inspector/hooks/local_link_connection.py b/ironic/drivers/modules/inspector/hooks/local_link_connection.py
index a3fb9fc71..17bc82f01 100644
--- a/ironic/drivers/modules/inspector/hooks/local_link_connection.py
+++ b/ironic/drivers/modules/inspector/hooks/local_link_connection.py
@@ -117,8 +117,10 @@ class LocalLinkConnectionHook(base.InspectionHook):
                 continue
 
             mac_address = iface['mac_address']
-            port = ironic_port.Port.get_by_address(task.context, mac_address)
-            if not port:
+            try:
+                port = ironic_port.Port.get_by_address(task.context,
+                                                       mac_address)
+            except exception.PortNotFound:
                 LOG.debug('Skipping LLDP processing for interface %s of node '
                           '%s: matching port not found in Ironic.',
                           mac_address, task.node.uuid)
